yamlfmt/yamlfmt.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/kylelemons/godebug/diff"
+	yaml "sigs.k8s.io/yaml/goyaml.v3"
+	"sigs.k8s.io/yaml/kyaml"
+)
+
+const (
+	fmtYAML  = "yaml"
+	fmtKYAML = "kyaml"
+)
+
+func main() {
+	help := flag.Bool("?", false, "print usage and exit")
+	diff := flag.Bool("d", false, "diff input files with their formatted versions")
```

Comment:
is built-in diff really worth pulling a new dep, instead of having people run `diff -u $file <(yamlfmt $file)`?

Reply:
I thought so.

I could exec `diff` against a temp file?

I could steal some simple line-diff algorithm and embed it here?

Reply:
We already dep on cmp but cmp.Diff is gross output.  I'll add a trivial diff function.